src/EventSubscriber/PartialMigrationSubscriber.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\helfi_api_base\EventSubscriber;
+
+use Drupal\Core\KeyValueStore\KeyValueFactoryInterface;
+use Drupal\migrate\Event\MigrateImportEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Resolve the state of the migration.
+ */
+final class PartialMigrationSubscriber implements EventSubscriberInterface {
+
+  public const PARTIAL_MIGRATE_KEY = 'partial_migrate_state';
+
+  public const PARTIAL_MIGRATE_LAST_FULL_KEY = 'partial_migrate_previous_full';
+
+  public const PARTIAL_MIGRATE_INTERVAL = 604800;
+
+  /**
+   * State collection.
+   *
+   * @var Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * PartialMigrationSubscriber constructor.
```

Comment:
This should be something else to make this easier to refactor, like `Constructs a new instance`.